code/datums/diseases/advance/symptoms/skin.dm

```diff
@@ -181,3 +181,137 @@ BONUS
+
+
+/datum/symptom/skineggs/Activate(datum/disease/advance/A)
+	if(!..())
+		return
+	var/mob/living/M = A.affected_mob
+	var/list/diseases = list(A)
+	switch(A.stage)
+		if(5)
+			if(thresholds["alldisease"])
+				for(var/datum/disease/D in M.diseases)
+					if((D.spread_flags & DISEASE_SPREAD_SPECIAL) || (D.spread_flags & DISEASE_SPREAD_NON_CONTAGIOUS) || (D.spread_flags & DISEASE_SPREAD_FALTERED))
+						continue
+					if(D == A)
+						continue
+					diseases += D
+			new /obj/item/reagent_containers/food/snacks/eggsac(diseases, thresholds["eggsplosion"], thresholds["sneaky"], thresholds["bigheal"])
+
+/obj/item/reagent_containers/food/snacks/eggsac
+	name = "Fleshy Egg Sac"
+	desc = "A small Egg Sac which appears to be made out of someone's flesh!"
+	customfoodfilling = FALSE //Not Used For Filling
+	icon_state = "icons/obj/food/food/eggsac.dmi"
+	bitesize = 4
+	var/list/diseases = list()
+	var/sneakyegg
+	var/bigheal
+
+//Constructor
+/obj/item/reagent_containers/food/snacks/eggsac/New(var/list/disease, var/eggsplodes, var/sneaky, var/largeheal)
+	for(var/datum/disease/D in disease)
+		diseases += D
+	if(largeheal)
+		src.add_initial_reagents(list(/datum/reagent/medicine/bicaridine = 20, /datum/reagent/medicine/tricordrazine = 10))
+		src.reagents.add_reagent(/datum/reagent/blood, 10, diseases)
+		bigheal = TRUE
+	else
+		src.add_initial_reagents(list(/datum/reagent/medicine/bicaridine = 10, /datum/reagent/medicine/tricordrazine = 10))
+		src.reagents.add_reagent(/datum/reagent/blood, 15, diseases)
+	if(sneaky)
+		icon_state = "icons/obj/food/food/eggsac-sneaky.dmi"
+		sneakyegg = sneaky
+	if(eggsplodes)
+		addtimer(CALLBACK(src, .proc/eggsplode), 100 SECONDS)
+	if(LAZYLEN(diseases))
+		AddComponent(/datum/component/infective, diseases)
+
+
+/obj/item/reagent_containers/food/snacks/eggsac/proc/eggsplode()
+	var/i
+	for(i = 1, i <= rand(4,8), i++)
+		var/list/directions = GLOB.alldirs
+		var/obj/item/I = new /obj/item/reagent_containers/food/snacks/fleshegg(src.diseases, src.sneakyegg, src.bigheal)
+		var/turf/thrown_at = get_ranged_target_turf(I, pick(directions), rand(2, 4))
+		I.throw_at(thrown_at, rand(2,4), 4)
+
+/obj/item/reagent_containers/food/snacks/fleshegg
+	name = "Fleshy Egg"
+	desc = "An Egg which appears to be made out of someone's flesh!"
+	customfoodfilling = FALSE //Not Used For Filling
+	icon_state = "icons/obj/food/food/fleshegg.dmi"
+	bitesize = 1
+	var/list/diseases = list()
+
+/obj/item/reagent_containers/food/snacks/fleshegg/New(var/list/disease, var/sneaky, var/largeheal)
+	for(var/datum/disease/D in disease)
+		diseases += D
+	if(largeheal)
+		src.add_initial_reagents(list(/datum/reagent/medicine/bicaridine = 20, /datum/reagent/medicine/tricordrazine = 10))
+		src.reagents.add_reagent(/datum/reagent/blood, 10, diseases)
+	else
+		src.add_initial_reagents(list(/datum/reagent/medicine/bicaridine = 10, /datum/reagent/medicine/tricordrazine = 10))
+		src.reagents.add_reagent(/datum/reagent/blood, 15, diseases)
```

Comment:
and here

Reply:
Those are both prices that need to be called on an object, and SRC indicates the object being made as what is being called to.

Reply:
`src` is implied. You do not need to specify it in this case.